src/mtev_http.c

```diff
@@ -1804,6 +1811,7 @@ _http_construct_leader(mtev_http_session_ctx *ctx) {
         }
       }
     }
+    if (!cl_present && keys != (const char **)static_key_array) free(keys);
```

Comment:
Confirmed - if we remove that check, we crash.

I see you approved it; I'll go ahead and merge.